File: modules/countdown-activity/android/src/main/java/expo/modules/mywidgetlibrary/CountdownActivityModule.kt
package expo.modules.countdownactivity

import android.content.Context
import android.content.Intent
import android.os.Build
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class CountdownActivityModule : Module() {
    private val context: Context
        get() = requireNotNull(appContext.reactContext) { "React context is not available" }

    override fun definition() = ModuleDefinition {
        Name("CountdownActivityModule")

        Function("startLiveActivity") { endTime: Double, title: String, timerId: String ->
            val intent = Intent(context, TimerService::class.java).apply {
                action = TimerService.ACTION_START
                putExtra("endTime", endTime)
                putExtra("title", title)
                putExtra("timerId", timerId)
                putExtra("startTime", System.currentTimeMillis())
            }

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(intent)
            } else {
                context.startService(intent)
            }
        }

        Function("stopLiveActivity") { timerId: String ->
            val intent = Intent(context, TimerService::class.java).apply {
                action = TimerService.ACTION_STOP
                putExtra("timerId", timerId)
            }
            context.startService(intent)
        }
    }
}

File: modules/countdown-activity/ios/CountdownActivityModule.swift
import ExpoModulesCore

public class CountdownActivityModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CountdownActivityModule")

    Function("startLiveActivity") { (endTime: Double, title: String, timerId: String) in
      if #available(iOS 16.2, *) {
        let controller = ActivityController()
        controller.startLiveActivity(endTime: endTime, title: title, timerId: timerId)
      } else {
        print("Live Activities are not supported on this iOS version.")
      }
    }

    Function("stopLiveActivity") { (timerId: String) in
      if #available(iOS 16.2, *) {
        let controller = ActivityController()
        controller.stopLiveActivity(timerId: timerId)
      }
    }
  }
}